File: C/leet/t2.c
#include <stdio.h>
struct ListNode {
  int val;
  struct ListNode *next;
};

struct ListNode *addTwoNumbers(struct ListNode *l1, struct ListNode *l2) {
  struct ListNode *res;
  int debug = l1->val;
  int mlt = 1;
  while (l1->next) {
    printf("number is %d", debug);
    debug += l1->val * mlt;
    l1 = l1->next;
    mlt *= 10;
  }

  return res;
}

int main() {
  struct ListNode *l1, *l2;
  l2->val = 6;
  l1->val = 4;
  l1->next = l2;
  addTwoNumbers(l1, l2);
}

File: Rust/src/levels/leet/t2.rs
// Definition for singly-linked list.
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
    pub val: i32,
    pub next: Option<Box<ListNode>>,
}

impl ListNode {
    #[inline]
    fn new(val: i32) -> Self {
        ListNode { next: None, val }
    }
}
pub fn add_two_numbers(
    l1: Option<Box<ListNode>>,
    l2: Option<Box<ListNode>>,
) -> Option<Box<ListNode>> {
    if let Some(l1) = l1 {
        println!("...");
        while let Some(ref bnext) = l1.next {
            println!("val is {}", bnext.val)
        }
    }
    None
}

#[cfg(test)]
mod test {
    use super::{add_two_numbers, ListNode};

    #[test]
    fn test_add_two_numbers() {
        let mut l1 = Some(Box::new(ListNode::new(5)));
        println!("...");
        if let Some(ref mut node) = l1 {
            println!("...");
            node.next = Some(Box::new(ListNode::new(10)));
        }
        add_two_numbers(l1.clone(), None);
    }
}
